csrc/scheduler/reduction_utils.cpp

```diff
@@ -876,8 +900,12 @@ class PersistentBufferProjector {
       // `T2->T1->T7->T6->T4->T5` which has all root domain information.
       // See FusionBroadcastPersistentReduction_CUDA for an example
       dummy_outputs_.emplace_back(add(buffer_replicate, buffer));
-      ir_utils::replaceValInExprInputs(
-          use->definition(), buffer, buffer_replicate);
+      // avoid replacing the use with itself, see
+      // https://github.com/NVIDIA/Fuser/issues/1533
+      if (buffer != buffer_replicate) {
```

Comment:
Is this the right place to handle this uncommon case? `RecomputeTv::recompute` returns the input buffer, which means nothing is recomputed? At least in that case it shouldn't be necessary to add the dummy output, should it?

Reply:
(1) Yes. The direct cause is self-replacement, here it is directly avoided. We can also handle this in other places e.g. (a) before `auto buffer_replicate = RecomputeTv::recompute(buffer, producers);`, we make sure `buffer` is not in `producers`. or (b) within `replaceValInExprInputs(Expr* expr, Val* reference, Val* substitute)` make sure `reference` is different from `substitute`

(2) Yes, `RecomputeTv::recompute(buffer, producers)` will put `producers` to the clone map, since `buffer` is also one of the `producers`, it is already in the clone map. So nothing is recomputed, just returns the original input.

(3) You are right. Revised to make sure dummpy output is only added when the replacement actually happens.